src/hints.rs

```diff
@@ -138,6 +141,7 @@ impl<'a> From<&'a NotificationHint> for MessageItem {
             &NotificationHint::Y(value)                 => (Y              .to_owned(), MessageItem::Int32(value)),
             &NotificationHint::Urgency(value)           => (URGENCY        .to_owned(), MessageItem::Byte(value as u8)),
             &NotificationHint::Custom(ref key, ref val) => (key            .to_owned(), MessageItem::Str(val.to_owned ())),
+            &NotificationHint::CustomInt(ref key, val)  => (key            .to_owned(), MessageItem::Int32(val)),
```

Comment:
that's how far I got, I don't really wanna change the API dramatically, generics would do that so either this or patch dbus-rs with an extra `trait IntoMessageItem`. I think a box is a bit expensive.

Wait .. what about an extra enum that's like Message Item and implements a conversion to MessageItem?